sharktank/sharktank/ops/default_impls.py

```diff
@@ -304,16 +304,26 @@ def interpolate_default(
     )
 
 
-@layer_norm.override(Tensor, Tensor, Tensor)
 def layer_norm_default(input, weight, bias, *, eps):
     input = unbox_tensor(input)
-    weight = unbox_tensor(weight)
-    bias = unbox_tensor(bias)
+    if weight is not None:
+        weight = unbox_tensor(weight)
+    else:
+        weight = torch.ones(input.shape)
```

Comment:
Does this mean that we normalize over all values as one layer? I am curious if this is the natural default behavior. I guess the other extreme is each element on its own, but this does not seem useful.

Reply:
`torch.ones` would create a tensor with a default dtype. It is desirable to deduce a dtype either from the bias or from the incoming input.

Reply:
Nice catch on dtype. I don't really understand why this is used, but it is used in the reference implementation.